src/test/crimson/test_perfcounters.cc

```diff
@@ -0,0 +1,59 @@
+#include <pthread.h>
+#include <iostream>
+#include <fmt/format.h>
+#include "common/Formatter.h"
+#include <seastar/core/app-template.hh>
+#include <seastar/core/sharded.hh>
+#include "common/perf_counters.h"
+#include "crimson/common/perf_counters_service.h"
+
+using ceph::common::PerfCountersService;
+enum {
+  perftest_first = 1000000,
+  perftest_count,
+  perftest_last,
+};
+
+static seastar::future<> test_perfcounters(){
+  return ceph::common::sharded_pfservice().start().then([] {
+    return ceph::common::sharded_pfservice().invoke_on_all([] (auto& s){
+      PerfCounters *perf_logger;
```

Comment:
could just remove this definition and put `auto perf_logger = plb.create_perf_counters();`, or define it close to where it is used for the first time.